python/InductionEq.py

```
@@ -0,0 +1,435 @@
+
+
+	if I_TI['time_integrator'] ==  'SSPRK33':
+        # 'SSPRK33_1', 'SSPRK33_2a', 'SSPRK33_2b', 'SSPRK33_3a', 'SSPRK33_3b'};
+        # calc_curlB_rho = {'calc_curlB_rho_1_3args', 'calc_curlB_rho_2_3args', 'calc_curlB_rho_3_3args'};
+		RK_Step = [];
+		if I_IEq['hall_term'] ==  'USE_HALL':
+			RK_Step.append('calc_curlB_rho_1_3args')
+		if I_RunOps['variable_u']:
+			RK_Step.append('calc_u_3_args')
+		RK_Step.append('SSPRK33_1')
+
+		RK_Step.append('SSPRK33_2a')
+		if I_IEq['hall_term']== 'USE_HALL':
+			RK_Step.append('calc_curlB_rho_2_3args')
+		if I_RunOps['variable_u']:
+			RK_Step.append('calc_u_3_args')
+		RK_Step.append('SSPRK33_2b')
+
+		RK_Step.append('SSPRK33_3a')
+		if I_IEq['hall_term'] ==  'USE_HALL':
+			RK_Step.append('calc_curlB_rho_3_3args')
+		if I_RunOps['variable_u']:
+			RK_Step.append('calc_u_3_args')
+		RK_Step.append('SSPRK33_3b')
+
+		RK_Step.append('calc_time_3_args')
+
+		time_integrator_num_fields = 3;
+
+# switch time_integrator_num_fields
+	if time_integrator_num_fields == 2:
+		IEq_fields = {'field_b':field_b, \
+		'field_b2': field_b2, \
+		'field_curlB_rho': field_curlB_rho, \
+		'field_u': field_u, \
+		'field_rho':field_rho, \
+		'current_time':current_time, \
+		'field_divB': field_divB, \
+		'norm2_output':norm2_output, \
+		'energy': energy, \
+		'L2error_B': L2error_B, \
+		'L2error_divB': L2error_divB}
+	elif time_integrator_num_fields == 3:
+		IEq_fields = {'field_b': field_b, \
+		'field_b2': field_b2, \
+		'field_b3': field_b3, \
+		'field_curlB_rho': field_curlB_rho, \
+		'field_u': field_u, \
+		'field_rho': field_rho, \
+		'current_time': current_time, \
+		'field_divB': field_divB, \
+                'norm2_output': norm2_output,\
+                'energy': energy,\
+                'L2error_B': L2error_B, \
+                'L2error_divB': L2error_divB}
+	else:
+        	print('Wrong value of time_integrator_num_fields =' + str( time_integrator_num_fields))
```

This should throw an exception.